### PR TITLE
[develop] web-pipeline updated

### PR DESCRIPTION
**Pull request by star-e**
updated native pipeline

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos-robot run test cases

File: native/cocos/base/TypeDef.h [hunk 1]
```diff
@@ -46,36 +46,36 @@ using index_t = int32_t;
...
+    constexpr T operator+(const T lhs, const T rhs) { return static_cast<T>(static_cast<std::underlying_type<T>::type>(lhs) + static_cast<std::underlying_type<T>::type>(rhs)); } \
+    constexpr T operator+(const T lhs, bool rhs) { return static_cast<T>(static_cast<std::underlying_type<T>::type>(lhs) + rhs); }                                                \
+    constexpr void operator|=(T &lhs, const T rhs) { lhs = static_cast<T>(static_cast<std::underlying_type<T>::type>(lhs) | static_cast<std::underlying_type<T>::type>(rhs)); }   \
+    constexpr void operator&=(T &lhs, const T rhs) { lhs = static_cast<T>(static_cast<std::underlying_type<T>::type>(lhs) & static_cast<std::underlying_type<T>::type>(rhs)); }   \
+    constexpr void operator^=(T &lhs, const T rhs) { lhs = static_cast<T>(static_cast<std::underlying_type<T>::type>(lhs) ^ static_cast<std::underlying_type<T>::type>(rhs)); }   \
+    constexpr bool hasFlag(const T flags, const T flagToTest) {                                                                                                                   \
```
Comment (minggo) [inline native/cocos/base/TypeDef.h:61]: Could you please format the code?

Comment (star-e) [reply native/cocos/base/TypeDef.h:61]: How shall i format them? I just changed `inline` to `constexpr`. 
I think they are in the same line.
They are a bit long in browser, so are separated in different lines.

Comment (star-e) [reply native/cocos/base/TypeDef.h:61]: I have formatted the `\`. They are aligned now.

**Review by minggo — APPROVED** [review]

**Comment by star-e** [comment]
@cocos-robot run test cases

**Pull request merged**